src/Transport/Config/ExtensionPoints/AzureServiceBusNamespacePartitioningSettings.cs

```diff
@@ -29,7 +29,7 @@ public AzureServiceBusNamespacePartitioningSettings(SettingsHolder settings)
             return this;
         }
 
-        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString)
+        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString, NamespacePurpose purpose = NamespacePurpose.Partitioning)
```

Comment:
I agree though that AddNamespace is not in the right place, it probably belongs in the root of the transportconfig (next to connectionstring)